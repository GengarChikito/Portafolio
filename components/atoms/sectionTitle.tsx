/**
 * Defino las 'props' de este componente. Solo necesito una:
 * - 'text': El título que quiero mostrar (ej. "Experiencia", "Educación").
 */
export interface SectionTitleProps { text: string; }

/**
 * Este es mi átomo 'SectionTitle'.
 * Lo creé para no tener que repetir los estilos del título (el <h2>)
 * en cada una de las secciones de mi portafolio.
 * Así, si quiero cambiar el estilo de todos los títulos, solo lo hago aquí.
 */
export function SectionTitle({ text }: SectionTitleProps) {
    return (
        // Es un <h2> simple con estilos en línea.
        // Uso una variable CSS (--border-strong) para el color del borde inferior,
        // así el color cambia automáticamente cuando cambio de tema (light/dark).
        // El #0048BA es solo un 'fallback' o valor por defecto si la variable no carga.
        <h2 style={{ marginBottom: 16, paddingBottom: 6, borderBottom: "2px solid var(--border-strong, #0048BA)" }}>
            {text}
        </h2>
    );
}