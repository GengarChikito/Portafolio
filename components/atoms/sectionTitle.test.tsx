import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import React from "react";
import { SectionTitle } from "./sectionTitle";

describe("SectionTitle", () => {
    const props = { text: "Experiencia Profesional" };

    it("renderiza un encabezado h2 con el texto correcto", () => {
        render(<SectionTitle {...props} />);
        const heading = screen.getByRole("heading", { level: 2, name: props.text });
        expect(heading).toBeInTheDocument();
        expect(heading).toHaveTextContent(props.text);
    });

    it("aplica estilos de margen, padding y borde inferior", () => {
        render(<SectionTitle {...props} />);
        const heading = screen.getByRole("heading", { level: 2 });
        const style = (heading as HTMLElement).style;

        expect(style.marginBottom).toBe("16px");
        expect(style.paddingBottom).toBe("6px");
        expect(style.borderBottom).toContain("2px solid");
        // Verifica que incluya la variable CSS o fallback
        expect(style.borderBottom).toContain("var(--border-strong, #0048BA)");
    });

    it("snapshot del componente", () => {
        const { container } = render(<SectionTitle {...props} />);
        expect(container).toMatchSnapshot();
    });
});
