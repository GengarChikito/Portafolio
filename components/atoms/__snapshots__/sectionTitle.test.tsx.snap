// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionTitle > snapshot del componente 1`] = `
<div>
  <h2
    style="margin-bottom: 16px; padding-bottom: 6px; border-bottom: 2px solid var(--border-strong, #0048BA);"
  >
    Experiencia Profesional
  </h2>
</div>
`;
